Python dependencies and build problems fix

ryanthecoder opened the pull request:
There was several package dependencies for jupyter missing and separately some packages did not switch from distutils to setuptools, instead some went to pep517 and pandas went to meson-build and this resulted in improper package info being generated and they were generating as version 0.0.0 which broke some of packaging import due to the system thinking that they were not meeting the minimum required version

ryanthecoder committed:
add missing jupyter dependencies

ryanthecoder committed:
bump pydantic

ryanthecoder committed:
add all the missing dependencies

ryanthecoder committed:
fix nbformat not versioning

ryanthecoder committed:
fix everything but pandas

ryanthecoder committed:
Fix pandas building

sfoster1 reviewed (approved):
Looks good if it all still fits in the FS

ryanthecoder committed:
rip out some 'optional' dependencies
remove the packages from buildroot, remove the dependencies and write some patches to edit the toml files so pip doesn't complain

ryanthecoder committed:
use custom kernel config to rip out unneeded modules

ryanthecoder committed:
add some config setting that hopefully get smoothie communication working again

ryanthecoder committed:
don't let uboot interrupt startup

ryanthecoder committed:
update config.txt

ryanthecoder committed:
add gpio tool to debug

ryanthecoder committed:
fix the MF ttyAMA0

sfoster1 reviewed (approved):
Hooray! Let's merge it!
